src/containers/Registration.tsx

```diff
@@ -0,0 +1,209 @@
+  padding: 2rem 0.75rem 2rem 1.7rem !important;
+  height: 5.5rem !important;
+  font-size: 1.5rem;
+`;
+
+const StyledFloatingLabel = styled(FloatingLabel)`
+  font-size: 1.5rem;
+  label {
+    padding: 2rem 1.7rem;
+  }
+  .form-floating > .form-control:focus ~ label,
+  .form-control:not(:placeholder-shown) ~ label,
+  .form-control:focus ~ label {
+    transform: scale(0.8) translateY(-1.5rem) translateX(0);
+    opacity: 0.65 !important;
+  }
+`;
+
+const StyledCounterLabel = styled.label`
+  text-align: right;
+  opacity: 0;
+  padding-right: 0;
+  margin-right: 0;
+  min-width: 124%;
+  transform-origin: 0 0;
+`;
+
+interface MyForm {
+  userName: string;
+  email: string;
+  dateOfBirth: Date;
+  password: string;
+  confirmPassword: string;
+}
+
+const schema = yup.object().shape({
+  userName: yup.string().required("Як вас звати?"),
+  email: yup
+    .string()
+    .email("Будь ласка, використовуйте формат електронної адреси")
+    .required("Введіть адресу електронної пошти"),
+  dateOfBirth: yup
+    .date()
+    .typeError("Вкажіть вашу дату народження")
+    .required("Вкажіть вашу дату народження"),
+  password: yup
+    .string()
+    .matches(
+      /^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#\$%\^&\*])(?=.{8,})/,
+      "Пароль має складатися не менше ніж з 8 символів з яких як мінімум: один великий, один маленький, одна цифра, один спеціальний символ"
+    )
+    .required("Введіть ваш пароль"),
+  confirmPassword: yup
+    .string()
+    .oneOf([yup.ref("password"), null], "Пароль має співпадати"),
+});
+
+const Registration = () => {
+  const [show, setShow] = useState(false);
+
+  const handleClose = () => setShow(false);
+  const handleShow = () => setShow(true);
+
+  const {
+    register,
+    handleSubmit,
+    watch,
+    reset,
+    formState: { errors },
+  } = useForm<MyForm>({
+    resolver: yupResolver(schema),
+  });
+
+  const submitForm = (data: MyForm) => {
+    console.log(data);
+    handleClose();
+    reset();
+  };
+
+  return (
+    <WrapperDiv>
+      <Button variant="primary" onClick={handleShow}>
+        Зареєструватися
+      </Button>
+
+      <Modal size="lg" centered show={show} onHide={handleClose}>
+        <Modal.Header closeButton style={{ margin: "0px" }}>
+          {" "}
+        </Modal.Header>
+        <StyledModalTitle>Створіть свій профіль</StyledModalTitle>
+        <Modal.Body>
+          <Form
+            style={{
+              paddingLeft: "2rem",
+              paddingRight: "2rem",
+            }}
+            onSubmit={handleSubmit(submitForm)}
+          >
+            <StyledFloatingLabel controlId="userNameInput" label="Ім'я">
+              <StyledFormControl
+                {...register("userName")}
+                type="text"
+                name="userName"
+                placeholder="Ім'я"
+                maxLength={50}
+              />
+              <StyledCounterLabel>
+                {watch("userName") ? watch("userName").length : 0} / 50
```

Comment:
add variable before return to use only one watch() call

Reply:
Done